compiler/rustc_hir/src/hir.rs

```diff
@@ -226,12 +223,12 @@ pub struct PathSegment<'hir> {
 
 impl<'hir> PathSegment<'hir> {
     /// Converts an identifier to the corresponding segment.
-    pub fn from_ident(ident: Ident) -> PathSegment<'hir> {
-        PathSegment { ident, hir_id: None, res: None, infer_args: true, args: None }
+    pub fn from_ident(ident: Ident, hir_id: HirId, res: Res) -> PathSegment<'hir> {
```

Comment:
```suggestion
    pub fn new(ident: Ident, hir_id: HirId, res: Res) -> PathSegment<'hir> {
```
With these two new parameters it's no longer "from ident".